File: FoodPickerWeb/src/models/RecipeIngredient.ts
import type { Recipe } from "./Recipe";
import type { Ingredient } from "./Ingredient";
import type { Unit } from "./Unit";

export interface RecipeIngredient{
    id: number;
    quantity: number;
    unitId: number;
    unit?: Unit;
    is_optional: boolean;
    recipeId: number;
    recipe?: Recipe;
    ingredientId: number;
    ingredient?: Ingredient;
}

File: FoodPickerAPI_EF/FoodPicker_DB_Schema/Entities/Recipe_ingredient.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodPickerDB_Schema.Entities
{
	public class Recipe_ingredient
	{
		[Key]
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }

		public required int Quantity { get; set; }

		public bool Is_optional { get; set; }

		public int UnitId { get; set; }

		public virtual Unit? Unit { get; set; }

		public int RecipeId { get; set; }

		public virtual Recipe? Recipe { get; set; }

		public int IngredientId { get; set; }

		public virtual Ingredient? Ingredient { get; set; }
	}
}
